core/trino-main/src/test/java/io/trino/operator/join/unspilled/TestLookupJoinPageBuilder.java

```diff
@@ -221,7 +221,7 @@ public long getJoinPosition(int position, Page page, Page allChannelsPage, long
         @Override
         public long getJoinPosition(int position, Page hashChannelsPage, Page allChannelsPage)
         {
-            throw new UnsupportedOperationException();
+            return -1;
```

Comment:
Why is this changing ?

Reply:
The tests in this class initialize `JoinProbe` without really using it. After introducing cache to JoinProbe, the LookupSource is being used in the constructor and invokes this method.